usaspending_api/awards/v2/data_layer/orm.py

```diff
@@ -370,8 +368,68 @@ def fetch_business_categories_by_legal_entity_id(legal_entity_id):
     return []
 
 
+def fetch_all_cfda_details(award):
+    queryset = TransactionFABS.objects.filter(transaction__award_id=award["id"],).values(
+        "cfda_number", "federal_action_obligation", "non_federal_funding_amount", "total_funding_amount")
+    cfdas = {}
+    for item in queryset:
+        # sometimes the transactions data has the trailing 0 in the CFDA number truncated, this adds it back
+        num = re.sub(r"^\d*\.\d\d$", "\\g<0>0", item["cfda_number"])
+        if cfdas.get(num):
+            cfdas.update(
+                {
+                    num: {
+                        "federal_action_obligation": cfdas[num]["federal_action_obligation"]
+                        + float(item["federal_action_obligation"] or 0),
```

Comment:
Because this is the page for an entire reward, are we sure that a float type will have enough precision to still handle individual cents when the total amount is in the tens (possibly hundreds) of billions?

Reply:
i mean i'm not sure, but i'll change it to `Decimal` if it makes you feel better

Reply:
I'm touched. I'd feel even better if you did it in all the other places too.